src/server.cpp

```diff
@@ -127,9 +127,13 @@ void *ServerThread::run()
 
 		u64 t0 = porting::getTimeUs();
 
-		const Server::StepSettings step_settings = m_server->getStepSettings();
+		const auto step_settings = m_server->getStepSettings();
 
 		try {
+			// see explanation inside
+			if (dtime > step_settings.steplen)
```

Comment:
(Would make more sense to compare with the steplen of the previous step. But probably doesn't matter in practice.)